astropy/time/core.py

```diff
@@ -2232,6 +2234,40 @@ def __add__(self, other):
     def __radd__(self, other):
         return self.__add__(other)
 
+    def __array_function__(self, function, types, args, kwargs):
+        """Wrap numpy functions.
+
+        Parameters
+        ----------
+        function : callable
+            Numpy function to wrap
+        types : iterable of classes
+            Classes that provide an ``__array_function__`` override. Can
+            in principle be used to interact with other classes. Below,
+            mostly passed on to `~numpy.ndarray`, which can only interact
+            with subclasses.
+        args : tuple
+            Positional arguments provided in the function call.
+        kwargs : dict
+            Keyword arguments provided in the function call.
+
+        Returns
+        -------
+        result: `~astropy.time.Time`
+        """
+        if function in TWOARG_HELPERS:
+            function_helper = TWOARG_HELPERS[function]
+            t, args = function_helper(*args)
+            jd = function(*args, **kwargs)
+            ret = t.__class__(jd, format='jd', scale=t.scale,
```

Comment:
This would not always work; you might consider having a `_result_as_time` method to deal with the output, like I did for `Quantity` and `Masked` (might be possible to factor out a bit from `_apply`

Reply:
I'm not sure what the correct approach here is - advice appreciated!

Reply:
I guess what is needed depends on part on the kind of overrides one does. For `np.linspace`, a complete override via a dispatched function probably makes most sense, in which case you just get a `Time` back here and can just return in.

Reply:
p.s. For `Quantity` and `Masked`, the `_result_as*` methods were mostly meant to reattach unit or mask. Here, that would make most sense if there are a set of numpy functions for which it is easier to just return `jd1` and `jd2`, and then create a new `Time` object similar to `self` (as is done at the end of `_apply` as well). But maybe worry about that when we get to functions that need it!

Reply:
Given the comment below about not using `jd` for `linspace`, I think we can cross this bridge when we come to it with another function